src/history/hs.c

```diff
@@ -108,10 +145,14 @@ __wt_hs_stats_update(WT_SESSION_IMPL *session)
     cstats = conn->stats;
 
     /*
-     * We have a cursor, and we need the underlying data handle; we can get to it by way of the
-     * underlying btree handle, but it's a little ugly.
+     * Get a history store cursor, we need the underlying data handle; we can get to it by way of
+     * the underlying btree handle, but it's a little ugly.
      */
-    dstats = ((WT_CURSOR_BTREE *)cache->hs_session[0]->hs_cursor)->btree->dhandle->stats;
+    if (F_ISSET(conn, WT_CONN_HS_OPEN) && __wt_hs_get_btree(session, &hs_btree) == 0 &&
+      hs_btree != NULL)
+        dstats = hs_btree->dhandle->stats;
+    else
+        return; /* Silently ignored the error in getting the history store btree. */
```

Comment:
I am not happy about ignoring a possible error here, but this being stats collection, it should be fine. Curious to know what the reviewers think about it.

Reply:
I'd rather not ignore the error. Could you change the function to return an error?

Reply:
Done: I had to change several functions to return int rather than void.